### PR TITLE
[stable/jenkins] Reorder README parameters into sections for readability

### PR DESCRIPTION
**Pull request by wmcdona89**
#### What this PR does / why we need it:
Reorder parameters in README to facilitate chart usage and maintenance.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:
There are a series of commits to facilitate the review. `values.yaml` could also be reordered.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/jenkins/README.md [hunk 1]
```diff
@@ -128,29 +128,69 @@ The following tables list the configurable parameters of the Jenkins chart and t
...
+| `master.JCasC.defaultConfig`      | Enables default Jenkins configuration via configuration as code plugin | `true`  |
+| `master.sidecars.configAutoReload` | Jenkins Config as Code auto-reload settings |                                   |
+| `master.sidecars.configAutoReload.enabled` | Jenkins Config as Code auto-reload settings (Attention: rbac needs to be enabled otherwise the sidecar can't read the config map) | `true`                                                      |
+| `master.sidecars.configAutoReload.image` | Image which triggers the reload | `kiwigrid/k8s-sidecar:0.1.144`           |
+| `master.sidecars.configAutoReload.env` | Environment variables for the Jenkins Config as Code auto-reload container  | Not set |
+| `master.testEnabled`              | Can be used to disable rendering test resources when using helm template | `true`                         |
+| Jenkins Configuration Files & Scripts |                                  |                                           |
+| `master.JCasC.configScripts`      | List of Jenkins Config as Code scripts | `{}`                                    |
```
Comment (torstenwalter) [inline stable/jenkins/README.md]: Should we move this below `master.JCasC.defaultConfig` to have all `JCasC` settings grouped together?

File: stable/jenkins/README.md [hunk 2]
```diff
@@ -128,29 +128,69 @@ The following tables list the configurable parameters of the Jenkins chart and t
...
+| `master.initScripts`              | List of Jenkins init scripts         | `[]`                                      |
+| `master.overwriteConfig`          | Replace init scripts and config w/ ConfigMap on boot  | `false`                  |
+| `master.credentialsXmlSecret`     | Kubernetes secret that contains a 'credentials.xml' file | Not set               |
+| `master.secretsFilesSecret`       | Kubernetes secret that contains 'secrets' files | Not set                        |
```
Comment (torstenwalter) [inline stable/jenkins/README.md:154]: Should we deprecate those?

File: stable/jenkins/README.md [hunk 3]
```diff
@@ -128,29 +128,69 @@ The following tables list the configurable parameters of the Jenkins chart and t
...
+| `master.jobs`                     | Jenkins XML job configs              | `{}`                                      |
+| `master.overwriteJobs`            | Replace jobs w/ ConfigMap on boot    | `false`                                   |
+| `master.customConfigMap`          | Deprecated: Use a custom ConfigMap   | `false`                                   |
+| `master.additionalConfig`         | Deprecated: Add additional config files | `{}`                                   |
```
Comment (torstenwalter) [inline stable/jenkins/README.md]: Does it make sense to move all deprecated settings to the end?

**Review by torstenwalter — COMMENTED** [review]
@wmcdona89 Thanks for suggesting this! It looks really good! One idea would be to split the table into multiple ones and use section titles as subheadings or keep it in one table and make those bold?

**Comment by scottrigby** [comment]
Testing infra was fixed, so last failure shouldn't happen again
/retest

**Comment by scottrigby** [comment]
Try retest after resolving merge conflict

File: stable/jenkins/README.md (hunk 1 above)
Comment (wmcdona89) [reply stable/jenkins/README.md]: good point. the `master.JCasC` parameters were in 3 different sections so I'll move them to a `Jenkins Configuration as Code` section.

File: stable/jenkins/README.md (hunk 3 above)
Comment (wmcdona89) [reply stable/jenkins/README.md]: yes. I'll make that change.

File: stable/jenkins/README.md (hunk 2 above)
Comment (wmcdona89) [reply stable/jenkins/README.md:154]: We can definitely start deprecating fields that have a JCasC alternative.

File: stable/jenkins/README.md (hunk 3 above)
Comment (wmcdona89) [reply stable/jenkins/README.md:158]: on 2nd thought...I think I'd rather leave the deprecated parameters in their respective sections. it's not obvious where a `Deprecated` section would go.

**Comment by wmcdona89** [comment]
@torstenwalter this is ready for another review

**Review by torstenwalter — APPROVED** [review]
/lgtm

**Pull request merged**